Tasks/AzurePowerShellV4/InitializeAz.ps1

```diff
@@ -12,11 +12,23 @@ $endpointObject =  ConvertFrom-Json  $endpoint
 $moduleName = "Az.Accounts"
 $environmentName = $endpointObject.environment
 
+. "$PSScriptRoot/Utility.ps1"
+Update-PSModulePathForHostedAgentLinux -targetAzurePs $targetAzurePs
+
 if($targetAzurePs -eq ""){
     $module = Get-Module -Name $moduleName -ListAvailable | Sort-Object Version -Descending | Select-Object -First 1
 }
 else{
-    $module = Get-Module -Name $moduleName -ListAvailable | Where-Object {$_.Version -eq $targetAzurePs} | Select-Object -First 1
+    $modules = Get-Module -Name $moduleName -ListAvailable
+    foreach ($moduleVal in $modules) {
+        $azModulePath = Split-Path (Split-Path (Split-Path $moduleVal.Path -Parent) -Parent) -Parent
```

Comment:
please add a comment here explaining why we are doing this split-path, otherwise it will be hard for someone else to understand this.